File: tp2/ej4/ej.pas
{ejemplo de programa que compila pero que tiene error de tipos en tiempo
 de ejecucion. El error ocurre cuando el input es un string, o algun valor
 que no sea entero}
Program Ej;
Var x : Integer;
begin
  WriteLn('Enter a number: ');
  ReadLn(x);
  WriteLn(x);
end.

File: tp7/ej1/ej.c
#include<stdio.h>

char c;
int hasError;
float n1, n2, answer;

int main(){
  hasError = 0;
  answer = 0;
  printf("Enter a char: ");
  scanf("%c",&c);
  printf("Enter a number: ");
  scanf("%f",&n1);
  printf("Enter a number: ");
  scanf("%f",&n2);

  switch(c){
    case '+':
      answer = n1 + n2;
      break;
    case '-':
      answer = n1 - n2;
      break;
    case '*':
      answer = n1 * n2;
      break;
    case '/':
      if(n2 == 0){
        hasError = 1;
      }
      
      if(!hasError){
        answer = n1 / n2;
      }
      break;
    default:
      hasError = 1;
      break;
  }

  if(hasError){
    printf("Error!\n");
  }else{
    printf("The answer is %f \n",answer);
  }

  return 0;
}
